src/main/java/com/exactpro/th2/crawler/Crawler.java

```diff
@@ -224,6 +269,44 @@ public Duration process() throws IOException, UnexpectedDataProcessorException {
         return Duration.of(sleepTime, ChronoUnit.MILLIS);
     }
 
+    private List<String> requestSessionAliases() {
+        List<String> messageStreams = dataProviderService.getMessageStreams(Empty.getDefaultInstance()).getListStringList();
+
+        LOGGER.debug("Got the following message streams from data provider: {}", String.join(", ", messageStreams));
+
+        return messageStreams;
+    }
+
+    private Collection<String> matchRequestedSessionAliases() {
+        List<String> res = new ArrayList<>();
+        List<String> sessionAliases = requestSessionAliases();
+
+        if (sessionAliasPatterns != null) {
```

Comment:
Please, check `sessionAliasPatterns` for `null` first. If it is null we don't need to create the `res` aliases and call the `requestSessionAliases` method